spacy_wordnet/wordnet_domains.py

```diff
@@ -68,16 +79,40 @@ def wordnet_synsets_for_domain(self, domains: List[str]):
         ]
 
     @staticmethod
-    def __find_synsets(token: Token, lang: str):
+    def __find_synsets(token: Token,
+                       lang: str,
+                       pos: Optional[Union[str, List[str]]] = None) -> List[Synset]:
+        if pos is None:
+            pos = []
+        elif isinstance(pos, str):
+            pos = [pos]
+        elif not isinstance(pos, list):
+            try:
+                pos = list(pos)
+            except TypeError:
+                raise TypeError("pos argument must be None, type str, or type list.")
+        
+        acceptable_pos = {"verb": VERB, "noun": NOUN, "adj": ADJ} # We can define this as a private class constant
+        # check if any element in `pos` is not in `acceptable_pos`
+        if set(pos).difference(acceptable_pos):
+            raise ValueError("pos argument must be a combination of 'verb', "
+                             "'noun', or 'adj'.")
+
+        token_pos: List[int] = [acceptable_pos[k] for k in pos]
+        if not token_pos:
+           token_pos = [token.pos]
         word_variants = [token.text]
-        if token.pos in [VERB, NOUN, ADJ]:
+        if token.pos in token_pos:
```

Comment:
I've seen a problem here:

If no `pos` param is provided, and `token.pos` is used as default, the `token.lemma_` is always included as a word variant. This could cause a weird behavior of the component

```suggestion
        if token.pos in (token_pos if pos else acceptable_pos.values()):
```

Reply:
Let me make sure I understand correctly.

If `pos is None` and `token.pos` is something other than `NOUN`, `VERB`, or `ADJ` --> this may cause unexpected `token.lemma_` values to be appended to the `word_variants` list.

Do you have a token I could test with?